internal/cmd/operator-sdk/alpha/config3alphato3/convert_config_3-alpha_to_3_test.go

```diff
@@ -0,0 +1,130 @@
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/format"
+)
+
+var _ = Describe("ConvertConfig3AlphaTo3", func() {
+	TruncatedDiff = false
+	DescribeTable("should return the expected config",
+		func(inputCfgStr, expectedCfgStr string) {
+			output, err := convertConfig3AlphaTo3([]byte(inputCfgStr))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(string(output)).To(Equal(expectedCfgStr))
+		},
+		Entry("basic", basicConfig, basicConfigExp),
+		Entry("no resources", noResourcesConfig, noResourcesConfigExp),
+		Entry("multiple resources", multipleAPIsConfig, multipleAPIsConfigExp),
+	)
+})
+
+const (
+	basicConfig = `domain: example.com
+layout: ansible.sdk.operatorframework.io/v1
+projectName: memcached-operator
+resources:
+- crdVersion: v1
+  group: cache
+  kind: Memcached
+  version: v1alpha1
+version: 3-alpha
+`
+	basicConfigExp = `domain: example.com
+layout: ansible.sdk.operatorframework.io/v1
+projectName: memcached-operator
+resources:
+- api:
+    crdVersion: v1
+    # TODO(user): Uncomment the below line if this resource's CRD is namespace scoped, else delete it.
+    # namespaced: true
+  # TODO(user): Uncomment the below line if this resource implements a controller, else delete it.
+  # controller: true
+  domain: example.com
+  group: cache
+  kind: Memcached
+  version: v1alpha1
+version: "3"
+`
+)
+
+const (
+	noResourcesConfig = `domain: example.com
+layout: ansible.sdk.operatorframework.io/v1
+projectName: memcached-operator
+version: 3-alpha
+`
+	noResourcesConfigExp = `domain: example.com
+layout: ansible.sdk.operatorframework.io/v1
+projectName: memcached-operator
+version: "3"
+`
+)
+
+const (
+	multipleAPIsConfig = `domain: example.com
+layout: ansible.sdk.operatorframework.io/v1
+projectName: memcached-operator
+resources:
+- crdVersion: v1
+  group: cache
+  kind: Memcached
+  version: v1alpha1
+  webhookVersion: v1
+- crdVersion: v1
+  group: cache
+  kind: MemcachedRS
+  version: v1alpha1
+version: 3-alpha
+`
+	multipleAPIsConfigExp = `domain: example.com
+layout: ansible.sdk.operatorframework.io/v1
+projectName: memcached-operator
+resources:
+- api:
+    crdVersion: v1
+    # TODO(user): Uncomment the below line if this resource's CRD is namespace scoped, else delete it.
+    # namespaced: true
+  # TODO(user): Uncomment the below line if this resource implements a controller, else delete it.
+  # controller: true
+  domain: example.com
+  group: cache
+  kind: Memcached
+  version: v1alpha1
+  webhooks:
+    # TODO(user): Uncomment the below line if this resource's webhook implements a conversion webhook, else delete it.
+    # conversion: true
+    # TODO(user): Uncomment the below line if this resource's webhook implements a defaulting webhook, else delete it.
+    # defaulting: true
+    # TODO(user): Uncomment the below line if this resource's webhook implements a validating webhook, else delete it.
+    # validation: true
+    webhookVersion: v1
```

Comment:
are we setting it as v1 always? Will not the biggest part of the user by using the deprecated APIs?
How we will know that?

Reply:
If `webhookVersion` is set in the 3-alpha config, that value is used. Otherwise, this command defaults it to v1 with a TODO above it.

Reply:
The webhookVersion is only added to the PROJECT file if/when we have the webhook. So, are we checking it or are we adding it for all options?

Then, if the is the scenario `If webhookVersion is set in the 3-alpha config` then, it means that the webhook was scaffold before which shows that would be the old version in the biggest part of the case. So, should not we do the opposite here?

Reply:
My previous statement was incorrect, updated.